### ISSUE TITLE
linkmetrics not working as I expect

### ISSUE THREAD

**Issue by purvisb** [openthread/openthread#6479]
**Describe the bug** 
Here's my outputs
```
> linkmetrics query fe80:0:0:0:1029:e885:cfe7:de20 single qmr
done
```
```
> linkmetrics mgmt fe80:0:0:0:1029:e885:cfe7:de20 enhanced-ack register qmr
Received Link Metrics Management Response from: fe80:0:0:0:1029:e885:cfe7:de20
Status: Other error
```
**To Reproduce**

Git commit id: de93711e223aade9867d4bf196258776be269b79
Using nrf52840-mdk
build all firmware with `USB=1 THREAD_VERSION=1.2 LINK_METRICS=1 JOINER=1 COMMISSIONER=1`

**Expected behavior** 
The same outputs as seen [here](https://github.com/openthread/openthread/blob/main/src/cli/README.md#linkmetrics-query-ipaddr-single-pqmr)

**Console/log output**
[linkmetric pcaps.zip](https://github.com/openthread/openthread/files/6332778/linkmetric.pcaps.zip)

**Comment by Irving-cl** [comment]
Hi @purvisb 

Thanks for raising the issue. There are a few problems here:

* Your commit `de93711` is a little old. There are some fixes to this issue after this commit. Please use the latest commit.
* Regarding `linkmetrics mgmt fe80:0:0:0:1029:e885:cfe7:de20 enhanced-ack register qmr
`, please note that according to spec, **at most two** metrics can be used in enhanced-ack probing. So you can do `linkmetrics mgmt fe80:0:0:0:1029:e885:cfe7:de20 enhanced-ack register qm` or `linkmetrics mgmt fe80:0:0:0:1029:e885:cfe7:de20 enhanced-ack register mr`, etc. These should work. We also mentioned it in our cli doc (Probably we should bold it):

![image](https://user-images.githubusercontent.com/12945188/115207991-b3150980-a12e-11eb-8000-2d5c1156303c.png)
* If you do the query from a child to a parent, it should work (with the 2 points I mentioned above). If you do the query from a parent to a child, there is an issue. I fixed the issue in #6483.

Please have a try and let me know :)

**Comment by jwhui** [comment]
Closing stale issue.
